src/brsTypes/components/RoSGNode.ts

```diff
@@ -213,6 +230,129 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
+            let indexValue = index.getValue();
+            let childrenSize = this.children.length;
+            if (numChildrenValue <= -1 && indexValue === 0) {
+                //short hand to return all children
+                return new RoArray(this.children.slice());
+            } else if (numChildrenValue <= 0 || indexValue < 0 || indexValue >= childrenSize) {
+                //these never return any children
+                return new RoArray([]);
+            } else {
+                //only valid cases
+                let endSliceIndex = Math.min(indexValue + numChildrenValue, childrenSize);
+                return new RoArray(this.children.slice(indexValue, endSliceIndex));
+            }
+
+            return new RoArray([]);
+        },
+    });
+
+    /* Finds a child node in the subject node list of children, and if found,
+    remove it from the list of children. The match is made on the basis of actual
+    object identity, that is, the value of the pointer to the child node.
+    return false if trying to remove anything that's not a node */
+    private removechild = new Callable("removechild", {
+        signature: {
+            args: [new StdlibArgument("child", ValueKind.Dynamic)],
+            returns: ValueKind.Boolean,
+        },
+        impl: (interpreter: Interpreter, child: BrsType) => {
+            var spliceIndex = -1;
+            if (child instanceof RoSGNode) {
+                for (let i = 0; i < this.children.length; i++) {
+                    let childNode = this.children[i];
+                    if (childNode === child) {
+                        spliceIndex = i;
+                        child.removeParent();
+                    }
+                }
+                if (spliceIndex >= 0) {
+                    this.children.splice(spliceIndex, 1);
+                }
+                return BrsBoolean.True;
+            }
+            return BrsBoolean.False;
+        },
+    });
+    /* If the subject node has been added to a parent node list of children,
+    return the parent node, otherwise return invalid.*/
+    private getparent = new Callable("getparent", {
+        signature: {
+            args: [],
+            returns: ValueKind.Dynamic,
+        },
+        impl: (interpreter: Interpreter) => {
+            return this.parent;
+        },
+    });
+
+    /* Creates a child node of type nodeType, and adds the new node to the end of the
+    subject node list of children */
+    private createchild = new Callable("createchild", {
+        signature: {
+            args: [new StdlibArgument("nodetype", ValueKind.String)],
+            returns: ValueKind.Object,
+        },
+        impl: (interpreter: Interpreter, nodetype: BrsString) => {
+            // currently we can't create a custom subclass object of roSGNode,
+            // so we'll always create generic RoSGNode object as child
+            var child = createNodeByType(nodetype);
```

Comment:
Could you please switch these `var`s to `let`s?  JavaScript's scoping of `var` is sometimes surprising, and `let` is basically always what you want 😃